aten/src/ATen/mps/MPSGuardImpl.h

```diff
@@ -0,0 +1,104 @@
+//  Copyright © 2022 Apple Inc.
+
+#pragma once
+#include <c10/core/impl/DeviceGuardImplInterface.h>
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+#include <ATen/mps/MPSStream.h>
+
+#ifdef __OBJC__
+#include <Foundation/Foundation.h>
+#include <Metal/Metal.h>
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#endif
+
+#include <ATen/Tensor.h>
+#include <c10/core/MemoryFormat.h>
+#include <c10/core/Storage.h>
+#include <c10/core/TensorImpl.h>
+#include <sys/_types/_size_t.h>
+#include <memory>
+#include <c10/core/UndefinedTensorImpl.h>
+#include <c10/util/intrusive_ptr.h>
+
+
+namespace at {
+namespace mps {
+
+struct TORCH_API MPSGuardImpl final : public c10::impl::DeviceGuardImplInterface {
```

Comment:
This appears to be a no-op guard; can you just inherit from NoOpDeviceGuardImpl?

Reply:
That's a good point, I have added a TODO, to move this to NoOpDeviceGuardImpl in future PR. Here is the issue filed:
https://github.com/pytorch/pytorch/issues/77170